app/src/main/java/com/lla/panicbutton/ui/fragments/RecordingsFragment.kt

```diff
@@ -31,11 +33,23 @@ class RecordingsFragment : Fragment(R.layout.fragment_recordings)  {
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
         setupRecyclerView()
+
+        uploadButton.setOnClickListener{
+            findNavController().navigate(R.id.action_recordingsFragment_to_setupFragment)
+            onBoardingButton()
+        }
     }
 
     private fun setupRecyclerView() = rvRecordings.apply {
         recordingAdapter = RecordingAdapter(recordingList)
         adapter = recordingAdapter
         layoutManager = LinearLayoutManager(requireContext())
     }
+
+    private fun onBoardingButton(){
+        val sharedPref = requireActivity().getSharedPreferences("onBoardingNext", Context.MODE_PRIVATE)
+        val editor = sharedPref.edit()
+        editor.putBoolean("setUpNext", true)
+        editor.apply()
```

Comment:
I think this setUpNext is supposed to track if this is the user's first time entering the app? maybe we can use another name for it, like isFirstTime and set that to true if it is and false if not? and function names should be actions like "toggleFirstTime" or "setsFirstTime" if I'm understanding the func correctly.

Reply:
Yeah, that's correct. Though, when I was setting this up, the logic was a bit weird because of how the fragments are. In this case, "setUpNext" is a key in shared preferences. When we have reached the recordings screen and the user clicks upload, it calls this function setting it to true meaning that the user has passed the onboarding phase already. Hence, every time the user clicks on upload, the done button will always be visible. I'm not too sure if it's setting it to false afterwards though, but it doesn't really matter I guess since it's always being set to true afterwards. Sorry for the bad explanation lol and yeah, I'll change the names.